boot.php

```diff
@@ -785,60 +785,85 @@ function get_scheme() {
 		return($this->scheme);
 	}
 
+	/**
+	 * @brief Retrieves the Friendica instance base URL
+	 *
+	 * Caches both SSL and non-SSL version for performance
+	 *
+	 * @param bool $ssl
+	 * @return string
+	 */
 	function get_baseurl($ssl = false) {
 
 		// Is the function called statically?
-		if (!is_object($this))
-			return(self::$a->get_baseurl($ssl));
+		if (!is_object($this)) {
+			return self::$a->get_baseurl($ssl);
+		}
 
-		$scheme = $this->scheme;
+		if (!isset($this->baseurl[$ssl ? 'https' : 'http'])) {
```

Comment:
It is a little bit confusing that you set the array key to "http" or "https" depending on the setting of the $ssl variable. I guess it would be better to simply use $ssl as a key. Thing is: the variable $ssl only switches between "http" and "https" under certain circumstances.